pkg/cloudprovider/amifamily/ami_test.go

```diff
@@ -0,0 +1,131 @@
+package amifamily
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+func TestAWS(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "AMISelector")
+}
+
+var (
+	defaultOwners = []*string{aws.String("self"), aws.String("amazon")}
+)
+
+var _ = Describe("AMI Selectors", func() {
+	It("should have default owners and use tags when prefixed aren't set", func() {
+
+		amiSelector := map[string]string{
+			"Name": "my-ami",
+		}
+		filters, owners := getFiltersAndOwners(amiSelector)
+		Expect(owners).Should(ConsistOf(defaultOwners))
+		Expect(filters).Should(ConsistOf([]*ec2.Filter{
+			{
+				Name:   aws.String("tag:Name"),
+				Values: aws.StringSlice([]string{"my-ami"}),
+			},
+		}))
+	})
+	It("should have default owners and use name when prefixed", func() {
+		amiSelector := map[string]string{
+			"aws::name": "my-ami",
+		}
+		filters, owners := getFiltersAndOwners(amiSelector)
+		Expect(owners).Should(ConsistOf(defaultOwners))
+		Expect(filters).Should(ConsistOf([]*ec2.Filter{
+			{
+				Name:   aws.String("name"),
+				Values: aws.StringSlice([]string{"my-ami"}),
+			},
+		}))
+	})
+	It("should not set owners when legacy ids are passed", func() {
+		amiSelector := map[string]string{
+			"aws::name": "my-ami",
+			"aws-ids":   "ami-abcd1234,ami-cafeaced",
+		}
+		filters, owners := getFiltersAndOwners(amiSelector)
+		Expect(owners).Should(BeNil())
+		Expect(filters).Should(ConsistOf([]*ec2.Filter{
```

Comment:
Sanity check on `ConsistOf()`, this does a deep comparison of the elements rather than comparing the addresses, right?

Reply:
Definitely (this was one of my concerns) but the tests simply wouldn't pass if this wasn't the case